packages/runtime/agent-scheduler/src/scheduler.ts

```diff
@@ -55,7 +55,7 @@ class AgentScheduler extends EventEmitter implements IAgentScheduler {
     public get IAgentScheduler() { return this; }
 
     private get clientId(): string {
-        if (!this.runtime.IFluidHandleContext.isAttached) {
+        if (this.runtime.attachState === AttachState.Detached) {
```

Comment:
was there an issue with moving `isAttached` into the runtime e.g. `this.runtime.isAttached`?

Reply:
I do not think data store should have isAttached - it's not clear when it should be true - when we are attaching or already attached? The only reason isAttached exists on data store is because it implements IFluidHandleContext and I'll remove it (through a series of changes to disambiguate dds & object routes)

IFluidHandleContext is slightly different, as here it tells whether we want to engage attachment mechanism or not, so it lumps both state into one. It's possible that we should replace isAttached to attachState even here, i'm not sure.